### PR TITLE
Add release triplets

### PR DESCRIPTION
**Pull request by autoantwort**
Adds more release triplets.

### CODE REVIEW

**Review by FrankXie05 — APPROVED** [review]

**Comment by xiaozhuai** [comment]
Would you please also add `x86-windows-static-release`?

File: triplets/community/arm64-osx-release.cmake [hunk 1]
```diff
@@ -0,0 +1,8 @@
+set(VCPKG_TARGET_ARCHITECTURE arm64)
+set(VCPKG_CRT_LINKAGE dynamic)
+set(VCPKG_LIBRARY_LINKAGE static)
+
+set(VCPKG_CMAKE_SYSTEM_NAME Darwin)
+set(VCPKG_OSX_ARCHITECTURES arm64)
```
Comment (vicroms) [inline triplets/community/arm64-osx-release.cmake:6]: This line should not be needed. If this is actually needed, then we should fix our toolchain.

**Review by vicroms — CHANGES_REQUESTED** [review]

File: triplets/community/arm64-osx-release.cmake (hunk 1 above)
Comment (autoantwort) [reply triplets/community/arm64-osx-release.cmake:6]: The variable is only used in `vcpkg/scripts/cmake/vcpkg_configure_make.cmake`

Comment (autoantwort) [reply triplets/community/arm64-osx-release.cmake:6]: The variable was introduced in #6275 but as far as I see it was not used in that PR 😕

**Review by vicroms — APPROVED** [review]

**Pull request merged**